src/prefect/environments/storage/docker.py

```diff
@@ -108,7 +106,7 @@ def __init__(
         self.files = files or {}
         self.flows = dict()  # type: Dict[str, str]
         self._flows = dict()  # type: Dict[str, "prefect.core.flow.Flow"]
-        self.base_url = base_url or default_url
+        self.base_url = os.environ.get("DOCKER_HOST", base_url or default_url)
```

Comment:
I think that if a user provides a `base_url` it should be respected; so in this case we should change this line to:
```suggestion
        self.base_url = base_url or os.environ.get("DOCKER_HOST", default_url)
```